src/redux/reducers/api.ts

```diff
@@ -0,0 +1,44 @@
+import { FoundItem, Item } from '../../types/types';
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+import { URL } from '../../constants/constants';
+
+const api = createApi({
+  reducerPath: 'api',
+  baseQuery: fetchBaseQuery({ baseUrl: URL }),
+  endpoints: (build) => ({
+    getItems: build.query<Item[], string>({
+      queryFn: async () => {
+        try {
+          const response = await fetch(`${URL}?page=1&limit=100`);
+          return { data: (await response.json()).data };
+        } catch (e) {
+          return { error: { status: 404, data: 'Server Error' } };
+        }
+      },
+    }),
```

Comment:
I assume, you don't need to check these cases because if we have error for some reason, we will receive it in useGetItemsQuery and then we can use it like error or isError.